src/main/java/io/dropwizard/primer/auth/filter/PrimerAuthAnnotationFilter.java

```diff
@@ -64,12 +64,11 @@ public PrimerAuthAnnotationFilter(final PrimerConfigurationHolder configHolder,
     @Override
     @Metered(name = "authorize")
     public void filter(ContainerRequestContext requestContext) throws IOException {
-        // Do not proceed further with Auth if its disabled or whitelisted
-        if (!isEnabled() || isWhitelisted())
-            return;
 
         Optional<String> token = getToken(requestContext);
         if (!token.isPresent()) {
+            if (!isEnabled() || isWhitelisted())
```

Comment:
Just preempt isEnabled() check to top of method. That is more optimal and will keep the code DRY

Reply:
Done